### PR TITLE
dts: msm8937: Add Motorola Moto G5 (cedric)

### PR DESCRIPTION
**Pull request by ungeskriptet**
motorola-cedric_lk2nd

### CODE REVIEW

**Comment by rw-r-r-0644** [comment]
Does this work with the available kernels for cedric? I was looking at this yesterday (ended up accidentally duplicating part of the work before finding this PR :P)
With a very similar patch, loading a boot.img fails with the following error:
```
[320] Found boot image: hd1p54p0 : /mnt/boot.img
[450] DTB Total entry: 5, DTB version: 2
[450] ERROR: Unable to find suitable device tree for device (294/0x00010000/0x00010008/0)
[460] ERROR: Getting device tree address failed
[480] DTB offset is incorrect, kernel image does not have appended DTB
[490] Device info 0x00000126/00010000/0x00010008/0, pmic 0x10019/0x20011/0x0/0x0
[500] dtb not found
[500] Unable to load boot.img from ext2. Continuing legacy boot
[500] Unable to locate /bootselect partition
[510] ERROR: Invalid boot image header
[510] ERROR: Could not do normal boot. Reverting to fastboot mode.
[520] fs-boot: Can't open hd2
```
<details>
<summary>debugging informations:</summary>
The dev_tree_get_entry_info() call in aboot.c:copy_dtb fails; internally dev_tree_get_entry_info() calls platform_dt_absolute_match() to identify device trees compatible with the device. This always fails (matches no device trees) on the comparsion `(cur_dt_hlos_ddr == (target_get_hlos_subtype() & 0x700))`, since target_get_hlos_subtype() always returns 0 on this platform, while cur_dt_hlos_ddr has values 0x100, 0x200 ... 0x500 in the trees usually provided with cedric kernels.
</details>

cedric's stock aboot bootloader appears to use the dt entry board_hw_subtype field in a non-standard way unrelated to DDR size, compared against a custom hwrev field:
```C
...
          /* (pseucode snipped from decompilation of dev_tree_get_entry_info) */
          if (cur_dt_entry.board_hw_subtype <= target_hwrev()) {
              ...
          }
...
```

To make loading work with normal kernels, a custom dtb matching function would be required;
clearing bits 0x700 of platform_subtype in  `qcom,board-id = <variant_id platform_subtype>;` *should* bypass this issue, but would only work in patched kernels

**Comment by rw-r-r-0644** [comment]
One remaining issue could be getting this to boot on devices with product id 0x41 (it may already work, my device is 0x42 so I have no way to test this; though looking at the dev_tree_get_entry_info pseudocode the stock bootloader appears to do a strict comparsion). It should be sufficient to add the remaining board ids in the dts:

```
	qcom,board-id = <0x42 0x8500>
			<0x42 0x8400>,
			<0x41 0x8300>,
			<0x41 0x8200>,
                        <0x41 0x8100>;
```
(a possible issue with this though is that for the next boot stage it will always pick a dts with the id of the first board id specified instead of the right one for the board, since lk2nd-device.c:update_board_id will update the board hardware id to the first one from the list)

Another possible board id could be
```
	qcom,board-id = <0x10008 0>;
```
Which should be the id of this board with the standard qcom naming scheme (instead of the one from motorola). The stock bootloader appears to accept this dts through a separate function

File: dts/msm8937-motorola-cedric.dts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+/include/ "msm8952.dtsi"
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id = <294 0>;
+	qcom,board-id = <0x42 0x8500>;
```
Comment (barni2000) [inline dts/msm8937-motorola-cedric.dts:10]: If I remember correctly cedric has more than one board-id it would be great to add them.

**Comment by barni2000** [comment]
This should be sent to msm8916-mainline/lk2nd

**Pull request closed without merging**